internal/api/version_check.go

```diff
@@ -22,7 +22,10 @@ func CheckSourcegraphVersion(version, constraint, minDate string) (bool, error)
 		return true, nil
 	}
 
-	buildDate := regexp.MustCompile(`^\d+_(\d{4}-\d{2}-\d{2})_[a-z0-9]{7}$`)
+	// This should match the version format returned by Sourcegraph, which uses the 12
+	// character abbreviated commit hash:
+	// https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/dev/ci/internal/ci/config.go?L96.
+	buildDate := regexp.MustCompile(`^\d+_(\d{4}-\d{2}-\d{2})_[a-z0-9]{12}$`)
```

Comment:
This needs to continue to accept the old format as well for backward compatibility. Since we don't actually use the revhash for anything, how about:

```suggestion
	buildDate := regexp.MustCompile(`^\d+_(\d{4}-\d{2}-\d{2})_[a-z0-9]{7,}$`)
```

Which should accept anything seven characters or longer. (Git is planning to transition to a 64 character ID format, so I don't really want to put a length constraint on there, provided it's the last element in the version.)

If we do this, let's also add test cases for the common lengths (7, 12, and 40, at least).

Reply:
That works for me, thanks for the suggestion!